packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -0,0 +1,125 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import ec2 = require('@aws-cdk/aws-ec2');
+import cdk = require('@aws-cdk/cdk');
+import { cloudformation, ClusterArn} from './ecs.generated';
+
+export interface ClusterProps {
+    /**
+     * A name for the cluster.
+     *
+     * @default CloudFormation-generated name
+     */
+    clusterName?: string;
+
+    /**
+     * The VPC where your ECS instances will be running
+     */
+    vpc: ec2.VpcNetworkRef;
+
+    /**
+     * Whether or not the containers can access the instance role
+     *
+     * @default false
+     */
+    containersAccessInstanceRole?: boolean;
+}
+
+// TODO: replace this with call to SSM, stored as "/aws/service/ecs/optimized-ami/amazon-linux/recommended"
+// https://raw.githubusercontent.com/awslabs/aws-cloudformation-templates/master/aws/services/ECS/EC2LaunchType/clusters/public-vpc.yml
+export const ECS_OPTIMIZED_AMI = new ec2.GenericLinuxImage({
+    'us-east-2': 'ami-028a9de0a7e353ed9',
+    'us-east-1': 'ami-00129b193dc81bc31',
+    'us-west-2': 'ami-00d4f478',
+    'us-west-1': 'ami-0d438d09af26c9583',
+    'eu-west-2': 'ami-a44db8c3',
+    'eu-west-3': 'ami-07da674f0655ef4e1',
+    'eu-west-1': 'ami-0af844a965e5738db',
+    'eu-central-1': 'ami-0291ba887ba0d515f',
+    'ap-northeast-2': 'ami-047d2a61f94f862dc',
+    'ap-northeast-1': 'ami-0041c416aa23033a2',
+    'ap-southeast-2': 'ami-0092e55c70015d8c3',
+    'ap-southeast-1': 'ami-091bf462afdb02c60',
+    'ca-central-1': 'ami-192fa27d',
+    'ap-south-1': 'ami-0c179ca015d301829',
+    'sa-east-1': 'ami-0018ff8ee48970ac3',
+    'us-gov-west-1': 'ami-c6079ba7',
+});
+
+// Needs to inherit from CloudFormationToken to make the string substitution
+// downstairs work. This is temporary, will go away in the near future.
+export class ClusterName extends cdk.CloudFormationToken {
+}
+
+export class Cluster extends cdk.Construct {
```

Comment:
Have you settled on class names for our distinct hierarchies yet? Is it going to be `Cluster` and `FargateCluster`?